firebase-case-study.md

```diff
@@ -0,0 +1,393 @@
+
+**All in one platform** - *It covers the entire application development cycle, and the platform contains features to build, release and monitor applications. Also, it provides tools to engage users and keep them using it*.
+
+**Focus on Front-End Development** - *The Firebase focuses on creating frontend code for mobile applications that makes application development convenient and helps to keep costs low, and it allows developers and companies to standardize the backend environment*.
+
+**Serverless** - *Firebase solves scaling up a database cluster. It provides an entirely serverless environment as it comes with a serverless architecture that requires users to pay based on requests. There is no requirement to manage or worry about server infrastructure*.
+
+**Machine Learning** - *It is advantageous as it allows developers to rely on machine learning. It has an ML kit with available APIs for different mobile platform features like text identification, face detection, image labeling, barcode scanning, and more*.
+
+**Monitor Errors** - *It can monitor both non-fatal and fatal errors, and reports are generated based on how errors affect users’ experience*.
+
+**Backups** - *The apps are protected from any possibility of data loss by relying on the automatic backup feature of this platform. Clicking the Backups tab of the Firebase Database gives one access to the auto-backup settings*.
+
+____________________________________________________________________________________________________________________
+
+## **Weaknesses:**
+
+![Weakneses](images/weaknesses.jpeg)
+
+- It’s not Open-Source
+- Firebase does not work in many countries
+- Only NoSQL databases are available
+- Slow Queries
+- Not all services are free to start
+- It’s not cheap and pricing is difficult to forecast
+- Runs only on Google Cloud
+- Dedicated Servers & Enterprise Support are not an option
+
+____________________________________________________________________________________________________________________
+
+**It’s not Open-Source** - *Firebase users can't modify Firebase source code because it is closed-source. Because of this issue, developers using Firebase end up using alternative platforms*.
+
+**Firebase does not work in many countries** - *Firebase is a subdomain from Google that only works in some countries*.
+
+**Only NoSQL databases are available** - *In Firebase, there is no option to use a relational database. There is a solution for this situation, and a user can craft a complex with Firebase. The implementation may be complex for a beginner using the platform. Firebase uses JSON, and there are virtually no SQL features, so migrating from the database will be more challenging*.
+
+**Slow Queries** - *Its specific limitations are a downfall of using it, like the frequency limited to 1 per second. No native aggregation queries and the maximum API request size is only 10 MiB. The document size limit is 1 MiB, and the Queries have slow performance*.
```

Comment:
```suggestion
**Slow Queries** - *There are various limitations applied to users depending on their tier, such as the frequency limitation to 1 per second **[ed: on which service?]**. Firebase does not support native aggregation queries and the maximum API request size is only 10 MiB. The document size limit is 1 MiB, and the Queries have slow performance*.
```

This needs to be qualified a little, is it Firestore that is limited in this way?